app/uk/gov/hmrc/tariffclassificationfrontend/connector/CasesConnector.scala

```diff
@@ -30,6 +30,11 @@ import scala.concurrent.Future
 @Singleton
 class CasesConnector @Inject()(configuration: AppConfig, client: HttpClient) {
 
+  def getOne(reference: String)(implicit hc: HeaderCarrier): Future[Option[Case]] = {
+    val url = configuration.bindingTariffClassificationUrl + "/cases/" + reference
+    client.GET[Option[Case]](url)
+  }
+
   def getGatewayCases(implicit hc: HeaderCarrier): Future[Seq[Case]] = {
     val url = configuration.bindingTariffClassificationUrl + "/cases?queue_id=none&assignee_id=none&sort-by=elapsed-days"
```

Comment:
To improve: Maybe I miss this one, but use a URIBuilder to handle this lines would be great

Reply:
Potentially. @gokyo Might now if theres a Scala equivalent.